hbase-balancer/src/main/java/org/apache/hadoop/hbase/CatalogAccessor.java

```diff
@@ -0,0 +1,1071 @@
+      getCatalogHTable(connection, TableName.META_TABLE_NAME).getScanner(scan)) {
+      res = resultScanner.next();
+    }
+    if (res == null) {
+      scan = getCatalogScan(connection, 1);
+      scan.setFilter(rowFilter);
+      try (ResultScanner resultScanner =
+        getCatalogHTable(connection, TableName.ROOT_TABLE_NAME).getScanner(scan)) {
+        res = resultScanner.next();
+      }
+    }
+    return res;
+  }
+
+  /**
+   * Lists all of the regions currently in META.
+   * @param connection to connect with
+   * @param excludeOfflinedSplitParents False if we are to include offlined/splitparents regions,
+   *          true and we'll leave out offlined regions from returned list
+   * @return List of all user-space regions.
+   */
+  @VisibleForTesting
+  public static List<RegionInfo> getAllRegions(Connection connection,
+    boolean excludeOfflinedSplitParents) throws IOException {
+    List<Pair<RegionInfo, ServerName>> result;
+
+    result = getTableRegionsAndLocations(connection, null, excludeOfflinedSplitParents);
+
+    return getListOfRegionInfos(result);
+
+  }
+
+  /**
+   * Gets all of the regions of the specified table. Do not use this method to get meta table
+   * regions, use methods in RootTableLocator instead.
+   * @param connection connection we're using
+   * @param tableName table we're looking for
+   * @return Ordered list of {@link RegionInfo}.
+   */
+  public static List<RegionInfo> getTableRegions(Connection connection, TableName tableName)
+  throws IOException {
+    return getTableRegions(connection, tableName, false);
+  }
+
+  /**
+   * Gets all of the regions of the specified table. Do not use this method to get meta table
+   * regions, use methods in RootTableLocator instead.
+   * @param connection connection we're using
+   * @param tableName table we're looking for
+   * @param excludeOfflinedSplitParents If true, do not include offlined split parents in the
+   *          return.
+   * @return Ordered list of {@link RegionInfo}.
+   */
+  public static List<RegionInfo> getTableRegions(Connection connection, TableName tableName,
+    final boolean excludeOfflinedSplitParents) throws IOException {
+    List<Pair<RegionInfo, ServerName>> result =
+      getTableRegionsAndLocations(connection, tableName, excludeOfflinedSplitParents);
+    return getListOfRegionInfos(result);
+  }
+
+  private static List<RegionInfo>
+    getListOfRegionInfos(final List<Pair<RegionInfo, ServerName>> pairs) {
+    if (pairs == null || pairs.isEmpty()) {
+      return Collections.emptyList();
+    }
+    List<RegionInfo> result = new ArrayList<>(pairs.size());
+    for (Pair<RegionInfo, ServerName> pair : pairs) {
+      result.add(pair.getFirst());
+    }
+    return result;
+  }
+
+  /**
+   * This method creates a Scan object that will only scan catalog rows that belong to the specified
+   * table. It doesn't specify any columns. This is a better alternative to just using a start row
```

Comment:
I don't get the second sentence.

Reply:
I think it means it will retrieve all the columns (there are not columns selected explicitly).

I only changed one line in this code actually. It's pretty much a copy form https://github.com/apache/hbase/blob/master/hbase-balancer/src/main/java/org/apache/hadoop/hbase/MetaTableAccessor.java#L323. 

It seems if I keep both classes CatalogAccessor and MetaTableAccessor (one as a facade). The methods that got copied as part of the migration would look newly added. I wonder if I should just move these methods back to MetaTableAccessor and make CatalogAccessor the facade, it would make the diffs less confusing?